client/collector.go

```diff
@@ -0,0 +1,130 @@
+				ActiveChannels: activeChannels,
+				SpecID:         platformCollector.SpecId,
+			}
+
+			collectorInfos = append(collectorInfos, collectorInfo)
+		}
+
+		return collectorInfos, nil
+	} else if appType == "ship" {
+		shipappCollectors, err := c.ShipClient.ListCollectors(appID, appType)
+		if err != nil {
+			return nil, err
+		}
+
+		shipCollectorInfos := make([]types.CollectorInfo, 0, 0)
+		for _, shipappCollector := range shipappCollectors {
+			activeChannels := make([]types.Channel, 0, 0)
+			for _, shipappCollectorChannel := range shipappCollector.ActiveChannels {
+				activeChannel := types.Channel{
+					ID:   shipappCollectorChannel.Id,
+					Name: shipappCollectorChannel.Name,
+				}
+
+				activeChannels = append(activeChannels, activeChannel)
+			}
+			shipCollectorInfo := types.CollectorInfo{
+				AppID:          shipappCollector.AppId,
+				CreatedAt:      shipappCollector.CreatedAt,
+				Name:           shipappCollector.Name,
+				ActiveChannels: activeChannels,
+				SpecID:         shipappCollector.SpecId,
+			}
+
+			shipCollectorInfos = append(shipCollectorInfos, shipCollectorInfo)
+		}
+
+		return shipCollectorInfos, nil
+	}
+
+	return nil, errors.New("unknown app type")
+}
+
+func (c *Client) UpdateCollector(appID string, appType string, specID string, yaml string) (interface{}, error) {
+	if appType == "platform" {
+		return c.PlatformClient.UpdateCollector(appID, appType, specID, yaml)
+	} else if appType == "ship" {
+		return c.ShipClient.UpdateCollector(appID, appType, specID, yaml)
+	}
+
+	return nil, errors.New("unknown app type")
+}
+
+func (c *Client) UpdateCollectorName(appID string, appType string, specID string, name string) (interface{}, error) {
+	if appType == "platform" {
+		return c.PlatformClient.UpdateCollectorName(appID, appType, specID, name)
+	} else if appType == "ship" {
+		return c.ShipClient.UpdateCollectorName(appID, appType, specID, name)
+	}
+
+	return nil, errors.New("unknown app type")
+}
+
+// func (c *Client) CreateCollector(appID string, appType string, name string, yaml string) (*collectors.AppCollectorInfo, error) {
+func (c *Client) CreateCollector(appID string, appType string, name string, yaml string) (*collectors.AppCollectorInfo, error) {
+	if appType == "platform" {
+		return c.PlatformClient.CreateCollector(appID, appType, name, yaml)
+	} else if appType == "ship" {
+		return c.ShipClient.CreateCollector(appID, appType, name, yaml)
+	}
+
+	return nil, errors.New("unknown app type")
+}
+
+func (c *Client) GetCollector(appID string, appType string, specID string) (*collectors.AppCollectorInfo, error) {
+	if appType == "platform" {
+		return c.PlatformClient.GetCollector(appID, appType, specID)
+	} else if appType == "ship" {
+		return c.ShipClient.GetCollector(appID, appType, specID)
+	}
+
+	return nil, errors.New("unknown app type")
+
+}
+
+func (c *Client) PromoteCollector(appID string, appType string, specID string, channelIDs ...string) error {
+
+	if appType == "platform" {
```

Comment:
This is the only functions that needs to make distinction between platform and ship apps